pyemma/coordinates/transform/transformer.py

```diff
@@ -35,15 +35,89 @@
 __all__ = ['Transformer']
 __author__ = 'noe, marscher'
 
+class TransformerIteratorContext(object):
+
+    def __init__(self, stride=1, lag=0):
+        self._lag = lag
+        self.__init_stride(stride)
+
+    def __init_stride(self, stride):
+        self._stride = stride
+        if isinstance(stride, np.ndarray):
+            keys = stride[:, 0]
+            self._trajectory_keys, self._trajectory_lengths = np.unique(keys, return_counts=True)
+        else:
+            self._trajectory_keys = None
+        self._uniform_stride = TransformerIteratorContext.is_uniform_stride(stride)
+        if not self.uniform_stride and not self.is_stride_sorted():
+            raise ValueError("Currently only sorted arrays allowed for random access")
+
+    def ra_indices_for_traj(self, traj):
+        """
+        Gives the indices for a trajectory file (without changing the order within the trajectory itself).
+        :param traj:
+        :return:
+        """
+        assert not self.uniform_stride, "requested random access indices, but is in uniform stride mode"
+        return self._stride[self._stride[:, 0] == traj][:, 1] if traj in self.traj_keys else np.array([])
```

Comment:
I guess the documentation is not quite clear about this method.. The context object holds the Nx2 array (self._stride). The "traj" argument is then merely an index, not the actual trajectory and the return value is an np.array of indices for that specific trajectory(-index). 

The indices in stride are absolute per trajectory, there is no overlap, they always begin at 0. I mean that if trajectory "0" has length 10, then you (currently) can't assign the trajectory 0 index 11 and mean the first frame of trajectory "1".